compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -256,25 +256,13 @@ impl<'tcx> fmt::Display for FrameInfo<'tcx> {
             if tcx.def_key(self.instance.def_id()).disambiguated_data.data
                 == DefPathData::ClosureExpr
             {
-                write!(f, "inside closure")?;
+                write!(f, "inside closure")
             } else {
                 // Note: this triggers a `good_path_bug` state, which means that if we ever get here
                 // we must emit a diagnostic. We should never display a `FrameInfo` unless we
                 // actually want to emit a warning or error to the user.
-                write!(f, "inside `{}`", self.instance)?;
+                write!(f, "inside `{}`", self.instance)
             }
-            if !self.span.is_dummy() {
-                let sm = tcx.sess.source_map();
-                let lo = sm.lookup_char_pos(self.span.lo());
-                write!(
-                    f,
-                    " at {}:{}:{}",
-                    sm.filename_for_diagnostics(&lo.file.name),
-                    lo.line,
-                    lo.col.to_usize() + 1
-                )?;
-            }
-            Ok(())
```

Comment:
Ah I guess this is redundant... makes sense.

Reply:
in miri it isn't so... maybe I need to find a better way

Reply:
(just force pushed the last commit if you wanna have a look)

Reply:
Ah right, in Miri we don't always emit the span

https://github.com/rust-lang/rust/blob/2816fecd26eaa5cb8d3e0070d2d924b1ed284220/src/tools/miri/src/diagnostics.rs#L355-L359

but it probably makes more sense to make the code that decides not to print the span, compensate for that by adding location information to what it prints?